caddyhttp/push/handler_test.go

```diff
@@ -307,6 +308,58 @@ func TestMiddlewareShouldInterceptLinkHeaderPusherError(t *testing.T) {
 	comparePushedResources(t, expectedPushedResources, pushingWriter.pushed)
 }
 
+func TestMiddlewareShouldPushIndexFile(t *testing.T) {
+	// given
+	indexFile := "/index.html"
+	request, err := http.NewRequest(http.MethodGet, "/", nil) // Request root directory, not indexfile itself
+	if err != nil {
+		t.Fatalf("Could not create HTTP request: %v", err)
+	}
+
+	root := "."
+
+	middleware := Middleware{
+		Next: httpserver.HandlerFunc(func(w http.ResponseWriter, r *http.Request) (int, error) {
+			return 0, nil
+		}),
+		Rules: []Rule{
+			{Path: indexFile, Resources: []Resource{
+				{Path: "/index.css", Method: http.MethodGet},
+			}},
+		},
+		Root: http.Dir(root),
+	}
+
+	_, err = os.Create(root + indexFile)
```

Comment:
Use filepath.Join to ensure cross-platform compatibility, instead of straight concatenation.